docs/architecture.md

```diff
@@ -1,188 +1,264 @@
+It validates the requests by executing Kubewarden policies.
 
-At the bottom of the stack, Kubewarden's integrates with Kubernetes using the
-concept of [Dynamic Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
+Kubewarden integrates with Kubernetes using
+[Dynamic Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
 In particular, Kubewarden operates as a Kubernetes Admission Webhook.
-`policy-server` is the actual Webhook endpoint that is reached by Kubernetes
-API server to validate relevant requests.
+The `policy-server` is the Webhook endpoint called by the Kubernetes API server to validate requests.
 
-Kubernetes is made aware of the existence of Kubewarden's Webhook endpoints by
-`kubewarden-controller`. This is done by registering either
-a `MutatingWebhookConfiguration` or a `ValidatingWebhookConfiguration`
-object.
+<!--TODO:Have I got this next paragraph right?-->
 
-This diagram shows the full architecture overview of a cluster running
-the Kubewarden stack:
+Kubernetes discovers Kubewarden's Webhook endpoints using `kubewarden-controller`.
+This works by Kubewarden registering either a `MutatingWebhookConfiguration`
+or a `ValidatingWebhookConfiguration` object with Kubernetes.
+
+<!--TODO:@vicuad points out we need a paragraph about the Audit Scanner.-->
+
+The diagram shows the architecture of a cluster running the Kubewarden stack:
+
+<!--TODO:To be clear. The stack refers to the entire diagram rather that the
+stacked components for e.g., CAP.-->
+
+<!--TODO:Do we really need the full diagram at this point. Maybe start
+with the simpler one in the next section-->
 
 ![Full architecture](/img/architecture.png)
 
-## Journey of a Kubewarden policy
+## The journey of a Kubewarden policy
+
+The architecture diagram appears complex at first.
+This section covers it step by step.
 
-The architecture diagram from above can be intimidating at first, this
-section explains it step by step.
+### Default policy server
 
-### Default Policy Server
+On a new cluster, the Kubewarden components defined are:
 
-On a fresh new cluster, the Kubewarden components defined are its Custom
-Resource Definitions, the `kubewarden-controller` Deployment and a `PolicyServer`
-Custom Resource named `default`.
+<!--TODO:How are CRDs shown in the diagram-->
+
+- the Custom Resource Definitions (CRD)
+- the `kubewarden-controller` Deployment
+- a `PolicyServer` Custom Resource named `default`.
 
 ![Defining the first ClusterAdmissionPolicy resource](/img/architecture_sequence_01.png)
 
-`kubewarden-controller` notices the default `PolicyServer` resource and, as a result of that,
+<!--TODO:For the next paragraph. What type of architecture diagram is this? In the
+diagram, which is `policy-server`. The aquamarine box with 'Kubewarden Policy
+Server'? Is 'Deployment' capitalization correct? What does 'notices' mean
```

Comment:
> What type of architecture diagram is this? In the
diagram, which is `policy-server`

It's an informal one., roughly like [C4 in Mermaid](http://mermaid.js.org/syntax/c4.html). We should come up with a better diagram, plus maybe [sequence diagrams](http://mermaid.js.org/syntax/sequenceDiagram.html), but not for this PR.

> Is 'Deployment' capitalization correct?

Yes, it's customary to uppercase k8s resources to distinguish them.